### ISSUE TITLE
Python wrapper on windows with user access only

### ISSUE THREAD

**Issue by aqeelahmed168** [usnistgov/REFPROP-wrappers#37]
I am trying to use Pyhton wrapper on Windows 10. I do not have full access so I installed the Pyhton using anaconda.
After installing the package as suggested on readme file "pip install ctREFPROP". 
After that, I launched jupyter-notebook from the anaconda to test the code

import os, numpy as np
from ctREFPROP.ctREFPROP import REFPROPFunctionLibrary

if __name__=='__main__':
    # If needed...
    os.environ['RPPREFIX'] = r'D:\Code\REFPROP-cmake\build\10\Release\\'
    
    RP = REFPROPFunctionLibrary(os.environ['RPPREFIX'])
    print(RP.RPVersion())
    MOLAR_BASE_SI = RP.GETENUMdll(0,"MOLAR BASE SI").iEnum

    r = RP.REFPROPdll("PROPANE","PQ","T",MOLAR_BASE_SI, 0,0,101325, 0, [1.0])
    print(r.ierr, r.herr, r.Output[0])

# Finally, I get this message

ValueErrorTraceback (most recent call last)
<ipython-input-14-578d98f7ccdb> in <module>()
      6     os.environ['RPPREFIX'] = r'D:\Code\REFPROP-cmake\build\10\Release\\'
      7 
----> 8     RP = REFPROPFunctionLibrary(os.environ['RPPREFIX'])
      9     print(RP.RPVersion())
     10     MOLAR_BASE_SI = RP.GETENUMdll(0,"MOLAR BASE SI").iEnum

C:\Users\ahmeda\AppData\Local\Continuum\anaconda2\lib\site-packages\ctREFPROP\ctREFPROP.pyc in REFPROPFunctionLibrary(name, shared_extension)
    138                 raise ValueError('Too many loadable shared libraries were found in the folder "{name:s}"; obtained libraries were: {libs:s}.  You must provide an absolute path to the shared library you would like to load'.format(name=name, libs=str(list(good_so))))
    139     else:
--> 140         raise ValueError('"{name:s}" is neither a directory nor a file'.format(name=name))
    141 
    142     # Now load the library

ValueError: "D:\Code\REFPROP-cmake\build\10\Release\\" is neither a directory nor a file

Could you please see if this can be solved.

**Comment by ianhbell** [comment]
A couple of things:

  *   Please use code formatting block for code
  *   You should have set the path to your REFPROP installation into the environment variable RPPREFIX if it is not already set, that is why you get this error.  If your RPPREFIX is already set(it is by the installer), then you can comment out that line

**Comment by aqeelahmed168** [comment]
Hi, I commented the  line pointing to path of environment variable RPPREFIX.  I am using Python version 3.6.
The code I am using is shown below:

import os, numpy as np
from ctREFPROP.ctREFPROP import REFPROPFunctionLibrary

if __name__=='__main__':
    # If needed...
    #os.environ['RPPREFIX'] = r'D:\Code\REFPROP-cmake\build\10\Release\\'
    #os.environ['RPPREFIX'] = r'C:\\Program Files (x86)\\REFPROP\\REFPRP64.DLL'
    
    RP = REFPROPFunctionLibrary(os.environ['RPPREFIX'])
    print(RP.RPVersion())

    MOLAR_BASE_SI = RP.GETENUMdll(0,"MOLAR BASE SI").iEnum

    r = RP.REFPROPdll("PROPANE","PQ","T",MOLAR_BASE_SI, 0,0,101325, 0, [1.0])
    print(r.ierr, r.herr, r.Output[0])

#However, now I get this message

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-6-441ce50f22a9> in <module>()
     10     print(RP.RPVersion())
     11 
---> 12     MOLAR_BASE_SI = RP.GETENUMdll(0,"MOLAR BASE SI").iEnum
     13 
     14     r = RP.REFPROPdll("PROPANE","PQ","T",MOLAR_BASE_SI, 0,0,101325, 0, [1.0])

c:\users\ahmeda\refprop-wrappers\wrappers\python\ctypes\ctREFPROP\ctREFPROP.py in GETENUMdll(self, iFlag, hEnum)
   1497         herr = ct.create_string_buffer(255)
   1498 
-> 1499         if self._GETENUMdll is None: raise ValueError("The function GETENUMdll could not be loaded from the shared library.")
   1500         self._GETENUMdll(ct.byref(iFlag),hEnum,ct.byref(iEnum),ct.byref(ierr),herr,255,255)
   1501 

ValueError: The function GETENUMdll could not be loaded from the shared library.

**Comment by ianhbell** [comment]
The example there is for REFPROP version 10 beta.  If you need legacy functions, then see the MANUAL.TXT in your installation of REFPROP to see what functions are included in the DLL.  All functions in the DLL are exposed.  See also: https://github.com/usnistgov/REFPROP-wrappers/blob/master/wrappers/python/ctypes/ctREFPROP/ctREFPROP.py#L205-L338

*Please* use code blocks: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code

**Comment by aqeelahmed168** [comment]
Hi,  thanks for the response. I commented the line pointing to path of environment variable RPPREFIX. I am using Python version 3.6 and REFPROP version 9.1
The code I am using is shown below:

```python
import os, numpy as np
from ctREFPROP.ctREFPROP import REFPROPFunctionLibrary

if __name__=='__main__':
    # If needed...
    #os.environ['RPPREFIX'] = r'D:\Code\REFPROP-cmake\build\10\Release\\'
    #os.environ['RPPREFIX'] = r'C:\\Program Files (x86)\\REFPROP\\REFPRP64.DLL'
    
    RP = REFPROPFunctionLibrary(os.environ['RPPREFIX'])
    print(RP.RPVersion())

    MOLAR_BASE_SI = RP.GETENUMdll(0,"MOLAR BASE SI").iEnum

    r = RP.REFPROPdll("PROPANE","PQ","T",MOLAR_BASE_SI, 0,0,101325, 0, [1.0])
    print(r.ierr, r.herr, r.Output[0])
```

## However, now I get this message

ValueError Traceback (most recent call last)
in ()
10 print(RP.RPVersion())
11
---> 12 MOLAR_BASE_SI = RP.GETENUMdll(0,"MOLAR BASE SI").iEnum
13
14 r = RP.REFPROPdll("PROPANE","PQ","T",MOLAR_BASE_SI, 0,0,101325, 0, [1.0])

c:\users\ahmeda\refprop-wrappers\wrappers\python\ctypes\ctREFPROP\ctREFPROP.py in GETENUMdll(self, iFlag, hEnum)
1497 herr = ct.create_string_buffer(255)
1498
-> 1499 if self._GETENUMdll is None: raise ValueError("The function GETENUMdll could not be loaded from the shared library.")
1500 self._GETENUMdll(ct.byref(iFlag),hEnum,ct.byref(iEnum),ct.byref(ierr),herr,255,255)
1501

ValueError: The function GETENUMdll could not be loaded from the shared library.

**Comment by ianhbell** [comment]
You want the SETUPdll function to setup the DLL, followed by the call to SATT:

``` python
import os, numpy as np
from ctREFPROP.ctREFPROP import REFPROPFunctionLibrary

def NBP():
    RP = REFPROPFunctionLibrary(os.environ['RPPREFIX'])
    print(RP.RPVersion())
    RP.SETPATHdll(os.environ['RPPREFIX'])
    r = RP.SETUPdll(1,"PROPANE.FLD","HMX.BNC","DEF")
    assert(r.ierr == 0)
    print(RP.PQFLSHdll(101.325, 0, [1.0], 0))

if __name__=='__main__':
    NBP()
```

**Comment by ianhbell** [comment]
Added example to the README.md too

**Comment by dwaguan** [comment]
I met the same problem. Please help, THX!
Refprop 9.1, Win10 x64, installed crREFPROP. Run the following code:

```
import os, numpy as np
from ctREFPROP.ctREFPROP import REFPROPFunctionLibrary

def NBP():
    RP = REFPROPFunctionLibrary(os.environ['RPPREFIX'])
    print(RP.RPVersion())
    RP.SETPATHdll(os.environ['RPPREFIX'])
    r = RP.SETUPdll(1,"PROPANE.FLD","HMX.BNC","DEF")
    assert(r.ierr == 0)
    print(RP.PQFLSHdll(101.325, 0, [1.0], 0))

if __name__=='__main__':
    NBP()
```

AND I GOT THIS:
```
ValueError                                Traceback (most recent call last)
<ipython-input-31-48af8c17f75f> in <module>
     11 
     12 if __name__=='__main__':
---> 13     NBP()

<ipython-input-31-48af8c17f75f> in NBP()
      3 
      4 def NBP():
----> 5     RP = REFPROPFunctionLibrary(os.environ['RPPREFIX'])
      6     print(RP.RPVersion())
      7     RP.SETPATHdll(os.environ['RPPREFIX'])

~\Anaconda3\lib\site-packages\ctREFPROP\ctREFPROP.py in REFPROPFunctionLibrary(name, shared_extension)
    165     managed_handle = None # Not part of the manager, we are creating a standalone instance of REFPROP
    166     return REFPROPInstance(dll, managed_handle)
--> 167 
    168 class REFPROPInstance(object):
    169 

~\Anaconda3\lib\site-packages\ctREFPROP\ctREFPROP.py in __init__(self, dll, managed_handle)
    539 
    540         self.DEFAULT = self.GETENUMdll(0, 'DEFAULT').iEnum
--> 541         self.MOLE_SI = self.GETENUMdll(0, 'MOLE SI').iEnum
    542         self.MASS_SI = self.GETENUMdll(0, 'MASS SI').iEnum
    543         self.SI_WITH_C = self.GETENUMdll(0, 'SI WITH C').iEnum

~\Anaconda3\lib\site-packages\ctREFPROP\ctREFPROP.py in GETENUMdll(self, iFlag, hEnum)
   1555 
   1556         if self._GETENUMdll is None: raise ValueError("The function GETENUMdll could not be loaded from the shared library.")
-> 1557         self._GETENUMdll(ct.byref(iFlag),hEnum,ct.byref(iEnum),ct.byref(ierr),herr,255,255)
   1558 
   1559         return self._GETENUMdlloutput_tuple(iEnum.value,ierr.value,trim(herr.raw))

ValueError: The function GETENUMdll could not be loaded from the shared library.
```

**Comment by ianhbell** [comment]
I fixed this in version 0.8.1 (got ahead of myself with the nice fix of enumerated values for REFPROP 10, but forgot about REFPROP 9.1).  You can do:

pip install -U ctREFPROP

to upgrade.  Also, you should upgrade to REFPROP 10!
